### PR TITLE
 warmup and meal-sharing tasks for week-2 homework done

### PR DESCRIPTION
**Pull request by basafilm**

### CODE REVIEW

File: nodejs/week2/meal_sharing/src/backend/routes/meals.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+const meals = require('../data/meals.json');
+   //  Respond with the json for the meal with the corresponding id
+
+
+   router.get('/meals/:id',(req, res )=> {
+   let mealId = Number(req.params.id)
+   let mealWithId = meals.filter(meal => mealId === meal.id)
+   const mealAndId = mealWithId.map(m => m );
+
+   return res.send(JSON.stringify(mealAndId))
```
Comment (OlkaA) [inline nodejs/week2/meal_sharing/src/backend/routes/meals.js:12]: no need for converting mealAndId to a JSON string in this case

File: nodejs/week2/meal_sharing/src/backend/routes/meals.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+   let maxPriceReq =Number(req.query.maxprice)
+   const mealTitle = req.query.title
+   const createdAfter = (req.query.createdAfter)
+   const createdDate = Date.parse(createdAfter)
+   let limit = Number(req.query.limit)
+
+   const mealsQuery = meals.filter((meal) =>{
+      if (maxPriceReq) {
+         return maxPriceReq <= meal.price;
```
Comment (OlkaA) [inline nodejs/week2/meal_sharing/src/backend/routes/meals.js:34]: Here is` maxPriceReq <= meal.price;` works opposite of what expected according to task.
You need to get meals that has a price smaller than maxPrice.

File: nodejs/week2/meal_sharing/src/backend/routes/meals.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        
+      } else if (limit) {
+         return meal.id <= limit;
+
+      } else if (mealTitle) {
+         return meal.title.includes(mealTitle)
```
Comment (OlkaA) [inline nodejs/week2/meal_sharing/src/backend/routes/meals.js:40]: `meal.title.includes(mealTitle)` does not consider letter case. 
So if I write `title=neapolitan` I will not get this meal.

File: nodejs/week2/meal_sharing/src/backend/routes/meals.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+})
+
+// Respond with the json for all the meals
+router.use(function(req, res, next) {
+   for (var key in req.query)
+   { 
+     req.query[key.toLowerCase()] = req.query[key];
+   }
+   next();
+ });
```
Comment (OlkaA) [inline nodejs/week2/meal_sharing/src/backend/routes/meals.js:23]: What is the purpose of this block of the code?

File: nodejs/week2/warmup/index.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+const app = express()
+const port = 3000
+
+app.get('/', (req, res) => res.send('Node js week-2 howmwork'))
+
+
+
+app.get('/numbers/add', function (req, res) {
+const a  = Number(req.query.first)
+const b  = Number(req.query.second)
```
[the remark this answers is not in the document]
Comment (basafilm) [reply nodejs/week2/warmup/index.js:12]: Thanks Olka

File: nodejs/week2/warmup/package.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "yes",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "start": "node src/warmup/index.js",
+    "dev": "node src/warmup/index.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
[the remark this answers is not in the document]
Comment (basafilm) [reply nodejs/week2/warmup/package.json:9]: In the week2 I have two index.js ; one for meal sharing and another for warmup . They are located in different folder (warmup and meal_sheraing ) that is why I used warmup path.
I will check

File: nodejs/week2/meal_sharing/src/backend/routes/meals.js (hunk 1 above)
Comment (basafilm) [reply nodejs/week2/meal_sharing/src/backend/routes/meals.js:12]: ok

File: nodejs/week2/meal_sharing/src/backend/routes/meals.js (hunk 2 above)
Comment (basafilm) [reply nodejs/week2/meal_sharing/src/backend/routes/meals.js:34]: ok. I thought I have to max and above price .

File: nodejs/week2/meal_sharing/src/backend/routes/meals.js (hunk 4 above)
Comment (basafilm) [reply nodejs/week2/meal_sharing/src/backend/routes/meals.js:23]: apply for next query functions to insensitive letters , capital or lower case in the search query.

File: nodejs/week2/meal_sharing/src/backend/routes/meals.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
+const express = require('express')
+const router = express.Router()
+const meals = require('../data/meals.json');
+   //  Respond with the json for the meal with the corresponding id
+
+
+   router.get('/meals/:id',(req, res )=> {
+   let mealId = Number(req.params.id)
+   let mealWithId = meals.filter(meal => mealId === meal.id)
+   const mealAndId = mealWithId.map(m => m );
```
[the remark this answers is not in the document]
Comment (basafilm) [reply nodejs/week2/meal_sharing/src/backend/routes/meals.js:10]: I wanted to return all keys based on the searched Id. not only the ID .

File: nodejs/week2/warmup/package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (basafilm) [reply nodejs/week2/warmup/package.json:9]: you are right . I changed

File: nodejs/week2/meal_sharing/src/backend/routes/meals.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (basafilm) [reply nodejs/week2/meal_sharing/src/backend/routes/meals.js:10]: Work . I changed .

**Comment by basafilm** [comment]
> Hi :)
> Good implementation of the feedback.
> 
> But I think there appears new issue with req.query.title.
> ` const mealTitle =parseInt(req.query.title)` returns NaN
> Don't think you should use parseInt here at all.
> 
> But also without parseInt it does not work for me.
> Instead of using RegEx, you can convert both paramaters with string.toLowerCase().
> Like:
> `meal.title.toLowerCase().includes(mealTitle.toLowerCase());`

Hi Olka
It returns an empty arry.
I remember I no need using ReqEx or tolowercase( agin. because I already used in middleware function below:
router.use(function(req, res, next) {
   // to insensetive latters capital or lower cases 
   for (var key in req.query)
   { 
     req.query[key.toLowerCase()] = req.query[key];
   }
   next();
 });n

**Pull request merged**